cas-server-core-services/src/main/java/org/jasig/cas/services/RegisteredServiceAccessStrategySupport.java

```diff
@@ -0,0 +1,170 @@
+package org.jasig.cas.services;
+
+import org.jasig.cas.authentication.Authentication;
+import org.jasig.cas.authentication.AuthenticationContext;
+import org.jasig.cas.authentication.PrincipalException;
+import org.jasig.cas.authentication.principal.Principal;
+import org.jasig.cas.authentication.principal.Service;
+import org.jasig.cas.ticket.ServiceTicket;
+import org.jasig.cas.ticket.TicketGrantingTicket;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.validation.constraints.NotNull;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This is {@link RegisteredServiceAccessStrategySupport} that encapsulates common
+ * operations relevant to registered service access strategy and authorizations.
+ * This is a support utility class that acts as a façade around common authorization
+ * and access strategy presented in CAS.
+ *
+ * @author Misagh Moayyed
+ * @author Dmitriy Kopylenko
+ * @since 4.3.0
+ */
+public final class RegisteredServiceAccessStrategySupport {
+
+    /**
+     * The constant LOGGER.
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(RegisteredServiceAccessStrategySupport.class);
+
+    /**
+     * Instantiates a new Registered service access strategy support.
+     */
+    private RegisteredServiceAccessStrategySupport() {
+    }
+
+    /**
+     * Ensure service access is allowed.
+     *
+     * @param service           the service
+     * @param registeredService the registered service
+     */
+    public static void ensureServiceAccessIsAllowed(final Service service, final RegisteredService registeredService) {
+        if (registeredService == null) {
+            final String msg = String.format("Unauthorized Service Access. Service [%s] is not found in service registry.",
+                    service.getId());
+            LOGGER.warn(msg);
+            throw new UnauthorizedServiceException(UnauthorizedServiceException.CODE_UNAUTHZ_SERVICE, msg);
+        }
+        if (!registeredService.getAccessStrategy().isServiceAccessAllowed()) {
+            final String msg = String.format("Unauthorized Service Access. Service [%s] is not enabled in service registry.",
+                    service.getId());
+
+            LOGGER.warn(msg);
+            throw new UnauthorizedServiceException(UnauthorizedServiceException.CODE_UNAUTHZ_SERVICE, msg);
+        }
+    }
+
+    /**
+     * Ensure service access is allowed.
+     *
+     * @param service           the service
+     * @param registeredService the registered service
+     * @param authentication    the authentication
+     * @throws UnauthorizedServiceException the unauthorized service exception
+     * @throws PrincipalException           the principal exception
+     */
+    public static void ensureServiceAccessIsAllowed(@NotNull final Service service,
```

Comment:
Cant start with "is" since that implies a boolean return. I could go with `ensurePrincipalAllowedForThisService`.